src/pet.js

```diff
@@ -0,0 +1,84 @@
+const AGE_INCREMENT = 1; 
+const MAXIMUM_AGE = 30;
+
+const HUNGER_INCREMENT = 5;
+const HUNGER_DECREMENT = 3;
+const MINIMUM_HUNGER = 0;
+const HUNGER_THRESHOLD = 5;
+const MAXIMUM_HUNGER = 10;
+
+const FITNESS_DECREMENT = 3;
+const FITNESS_INCREMENT = 4;
+const MAXIMUM_FITNESS = 10;
+const FITNESS_THRESHOLD = 3;
+const MINIMUM_FITNESS = 0;
+
+const DEFAULT_CONDITION = "I feel great!";
+const DEATH_STATEMENT = 'Your pet is no longer alive :(';
+
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = MINIMUM_HUNGER;
+    this.fitness = MAXIMUM_FITNESS;
+    this.children = [];
+} 
+Pet.prototype = {
+    get isAlive() {
+      return this.age < MAXIMUM_AGE && this.hunger < MAXIMUM_HUNGER && this.fitness > MINIMUM_FITNESS;
+    }
+  };
+Pet.prototype.growUp = function() {
+    if (!this.isAlive) { 
+        throw new Error(DEATH_STATEMENT);
+        }
+    this.age += AGE_INCREMENT;
+    this.hunger += HUNGER_INCREMENT;
+    this.fitness -= FITNESS_DECREMENT;
+};
+Pet.prototype.walk = function() {
+    if (!this.isAlive) { 
+        throw new Error(DEATH_STATEMENT);
+        }
+   if((this.fitness + FITNESS_INCREMENT) <= MAXIMUM_FITNESS){
```

Comment:
When defining `walk()`, I believe you want to only give the Pet a fitness increment if its fitness is not already at a maximum level. If that's the case, shouldn't you only be comparing the current fitness level (`this.fitness`) against what could be its maximum level, without any initial increment (inside of the _if statement_)?

So, if the Pet's fitness levels are below maximum, the increment happens, otherwise, it stays as it is.

Reply:
I understand what you are saying here but if you compare before adding the increment then it allows fitness to go above maximum, unless there is something that i am missing.